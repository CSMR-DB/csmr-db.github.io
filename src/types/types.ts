import React from 'react'

export type AnyExoticRefComponent<T> = React.ForwardRefExoticComponent<
  T & React.RefAttributes<any>
>

export type AnyExoticRefTargets =
  | ((instance: any) => void)
  | React.MutableRefObject<any>
  | null
