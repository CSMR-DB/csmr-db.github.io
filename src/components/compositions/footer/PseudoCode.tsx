import React from 'react'
import styled, {
  StyledComponent,
  FlattenSimpleInterpolation,
  css,
} from 'styled-components'

import { StyledA } from '../../StyledLink'

const StyledPseudoCode: StyledComponent<'div', any, {}, never> = styled.div`
  margin-bottom: 1rem;

  & > p {
    margin: 0;
    text-align: left;
  }
`
interface IProgramLineProps {
  $padding?: number
}

const ProgramLine: StyledComponent<
  'p',
  any,
  IProgramLineProps,
  never
> = styled.p`
  margin: 0 !important;

  ${({ $padding: padding = 0 }: IProgramLineProps): FlattenSimpleInterpolation => css`
    padding-left: ${padding}rem;
  `}
`

function Credit(href: string, title: string): JSX.Element {
  return (
    <StyledA $color={'white'} href={href} target="blank">
      {title}
    </StyledA>
  )
}

const CREDITS: { [key: string]: JSX.Element } = {
  gatsby: Credit('https://www.gatsbyjs.org', 'Gatsby'),
  styledComponents: Credit(
    'https://www.styled-components.com/',
    'styled-components'
  ),
  firaCode: Credit('https://www.npmjs.com/package/firacode', 'Fira Code'),
  vsCodeIcons: Credit(
    'https://marketplace.visualstudio.com/items?itemName=vscode-icons-team.vscode-icons',
    'vscode-icons'
  ),
  adobeIcons: Credit('https://www.iconfinder.com/graphiqa', 'Graphiqa Studio'),
  prettier: Credit('https://prettier.io/', 'Prettier'),
  carbon: Credit('https://carbon.now.sh/', 'Carbon'),
  ghPages: Credit('https://www.npmjs.com/package/gh-pages', 'gh-pages'),
  gsap: Credit('https://greensock.com/gsap/', 'GSAP'),
  reactGsap: Credit('https://www.npmjs.com/package/react-gsap', 'react-gsap'),
  pageTransitions: Credit(
    'https://www.gatsbyjs.org/packages/gatsby-plugin-transitions/',
    'gatsby-plugin-transitions'
  ),
}

export function PseudoCode(): JSX.Element {
  return (
    <StyledPseudoCode>
      <ProgramLine>type Requirements = {`{ essence: Emoji[] }`}</ProgramLine>
      <ProgramLine>type DevWebsite = {`{ typescript: boolean }`}</ProgramLine>
      <ProgramLine>interface IWebsite {'{'}</ProgramLine>
      <ProgramLine $padding={1}>{`url: string`}</ProgramLine>
      <ProgramLine $padding={1}>{`visit: () => WebPage`}</ProgramLine>
      <ProgramLine>{'}'}</ProgramLine>
      <br />
      <ProgramLine>{`async function websitePromise({ essence }: Requirements): Promise<IWebsite> {`}</ProgramLine>
      <ProgramLine $padding={1}>
        return await start({CREDITS.gatsby})
      </ProgramLine>
      <ProgramLine $padding={2}>.then((dev: DevWebsite) ={'>'} </ProgramLine>
      <ProgramLine $padding={3}>pipe(</ProgramLine>
      <ProgramLine $padding={4}>
        addStyle({CREDITS.styledComponents}),
      </ProgramLine>
      <ProgramLine $padding={4}>useFont({CREDITS.firaCode}),</ProgramLine>
      <ProgramLine $padding={4}>
        addCodeIcons({CREDITS.vsCodeIcons}),
      </ProgramLine>
      <ProgramLine $padding={4}>
        addAdobeIcons({CREDITS.adobeIcons}),
      </ProgramLine>
      <ProgramLine $padding={4}>
        prettifyCodeUsing({CREDITS.prettier}),
      </ProgramLine>
      <ProgramLine $padding={4}>
        makeCodeImagesUsing({CREDITS.carbon}),
      </ProgramLine>
      <ProgramLine $padding={4}>
        useAnimationLibraries({CREDITS.gsap}, {CREDITS.reactGsap},{' '}
        {CREDITS.pageTransitions}),
      </ProgramLine>
      <ProgramLine $padding={4}>sprinkleWith(essence),</ProgramLine>
      <ProgramLine $padding={4}>deployUsing({CREDITS.ghPages})</ProgramLine>
      <ProgramLine $padding={3}>)(dev)</ProgramLine>
      <ProgramLine $padding={2}>)</ProgramLine>
      <ProgramLine>{`}`}</ProgramLine>
      <br />
      <ProgramLine>websitePromise({`{ essence: [ ‚ù§Ô∏è, ‚ú®, üéÜ ] }`})</ProgramLine>
      <ProgramLine $padding={1}>.then((website: IWebsite) ={'>'}</ProgramLine>
      <ProgramLine $padding={2}>website.visit())</ProgramLine>
      <ProgramLine $padding={1}>)</ProgramLine>
    </StyledPseudoCode>
  )
}
