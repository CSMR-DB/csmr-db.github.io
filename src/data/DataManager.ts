import { ITool, toolsProvider } from './objects/toolsProvider'
import { themeProvider } from './objects/themeProvider'
import { DefaultTheme } from 'styled-components'
import { IRouteObject, routesProvider } from './objects/routesProvider'
import { ICourse, coursesProvider } from './objects/coursesProvider'
import { IContact, contactsProvider } from './objects/contactsProvider'
import {
  dynamicIconsProvider,
  IDynamicIcons,
} from './graphql/dynamicIconsProvider'
import {
  dynamicImagesProvider,
  IDynamicImages,
} from './graphql/dynamicImagesProvider'
import {
  experienceDataProvider,
  IExperienceData,
} from './graphql/experienceDataProvider'
import {
  gameWallpapersProvider,
  IGameWallpapers,
} from './graphql/gameWallpapersProvider'
import {
  ISeriesPosters,
  seriesPostersProvider,
} from './graphql/seriesPostersProvider'
import { ISiteMetadata } from '../types/interfaces'
import {
  ISkillsetData,
  skillsetDataProvider,
} from './graphql/skillsetDataProvider'
import { siteMetadataProvider } from './graphql/siteMetadataProvider'
import {
  IGraphicDesignData,
  graphicDesignDataProvider,
} from './graphql/graphicDesignDataProvider'
import {
  IPhotographySquares,
  photographySquaresProvider,
} from './graphql/photographySquaresProvider'
import {
  programmingDataProvider,
  IProgrammingData,
} from './graphql/programmingDataProvider'
import {
  photographyDataProvider,
  IPhotographyData,
} from './graphql/photographyDataProvider'

export abstract class AStaticDataManager {
  static contacts: IContact[]
  static courses: ICourse[]
  static routes: IRouteObject[]
  static theme: DefaultTheme
  static tools: ITool[]
  static siteMetadata: ISiteMetadata
  static skillsetData: ISkillsetData
  static experienceData: IExperienceData
  static programmingData: IProgrammingData
  static graphicDesignData: IGraphicDesignData
  static photographyData: IPhotographyData
  static gameWallpapers: IGameWallpapers
  static photographySquares: IPhotographySquares
  static seriesPosters: ISeriesPosters
  static dynamicImages: IDynamicImages
  static dynamicIcons: IDynamicIcons
}

/**
 * **Static Data Manager**
 *
 * Uses an abstract rather than an interface to implement, because TypeScript expects implemented interface to define the shape of an instance, not the class itself. Currently, using an abstract is the only way to get around that, as I can refer to it as
 * `typeof AStaticDataManager` anywhere I use it.
 *
 * Provides all 'static' data to Components through 1 single source of truth. No more duplicate-y GrahpQL queries with hidden differences between 2 files, but 1 predictable response.
 *
 * Everything else should remain consistent & predictable throughout the entire website to provide the best UX it can. I don't really need instances or a Singleton. The StaticDataManager shouldn't exist at runtime anyway.
 *
 * @Note:
 * StaticDataManager can only be used inside React Components, due to the use of the `useStaticQuery` hook in the Data properties.
 *
 * @FIXME:
 * I am not 100% satisfied with the current API "inconsistencies" when destructuring the data, but alas. That's a priority item for the next revision.
 */
export class StaticDataManager implements AStaticDataManager {
  static get routes(): IRouteObject[] {
    return routesProvider()
  }

  static get contacts(): IContact[] {
    return contactsProvider()
  }

  static get theme(): DefaultTheme {
    return themeProvider()
  }

  static get tools(): ITool[] {
    return toolsProvider()
  }

  static get courses(): ICourse[] {
    return coursesProvider()
  }

  static get siteMetadata(): ISiteMetadata {
    return siteMetadataProvider()
  }

  static get skillsetData(): ISkillsetData {
    return skillsetDataProvider()
  }

  static get experienceData(): IExperienceData {
    return experienceDataProvider()
  }

  static get programmingData(): IProgrammingData {
    return programmingDataProvider()
  }

  static get graphicDesignData(): IGraphicDesignData {
    return graphicDesignDataProvider()
  }

  static get photographyData(): IPhotographyData {
    return photographyDataProvider()
  }

  static get gameWallpapers(): IGameWallpapers {
    return gameWallpapersProvider()
  }

  static get photographySquares(): IPhotographySquares {
    return photographySquaresProvider()
  }

  static get seriesPosters(): ISeriesPosters {
    return seriesPostersProvider()
  }

  static get dynamicImages(): IDynamicImages {
    return dynamicImagesProvider()
  }

  static get dynamicIcons(): IDynamicIcons {
    return dynamicIconsProvider()
  }
}
