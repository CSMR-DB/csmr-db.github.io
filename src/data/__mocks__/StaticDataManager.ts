import { ExperienceData } from './../graphql/experienceDataProvider'
import { DynamicImages } from './../graphql/dynamicImagesProvider'
import { DynamicIcons } from './../graphql/dynamicIconsProvider'
import { ProjectsData } from '../graphql/projectsDataProvider'
import { SkillsetData } from '../graphql/skillsetDataProvider'
import { GameWallpapers } from '../graphql/gameWallpapersProvider'
import { SeriesPosters } from '../graphql/seriesPostersProvider'
import {
  SiteMetadata,
  SkillsetAllMarkdownRemark,
  ImageSharpAllFiles,
  ProjectsAllMarkdownRemark,
  ProjectMarkdownEdge,
  SkillsetMarkdownEdge,
  ExperienceAllMarkdownRemark,
} from '../../types/graphql.types'
import { PhotographySquares } from '../graphql/photographySquaresProvider'
import { PhotographyData } from '../graphql/photographyDataProvider'
import { Tool } from '../objects/toolsProvider'
import { AStaticDataManager } from '../StaticDataManager'
import { RouteObject } from '../objects/routesProvider'
import { Course } from '../objects/coursesProvider'
import { Contact } from '../objects/contactsProvider'
import { DefaultTheme } from 'styled-components'
import { mockSVGs } from './tools.mocks'

const mockImageCollection: ImageSharpAllFiles = {
  edges: [
    {
      node: {
        name: 'Photo',
        relativePath: '/photo.jpg',
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: '',
            src: '/photo.jpg',
            srcSet: '/photo.jpg',
          },
        },
      },
    },
  ],
}

const mockProgrammingSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'React',
          excerpt: 'I love JSX',
          favorite: true,
          icon: '',
          skillColor: 'red',
          time: 1337,
        },
        generatedRoute: '/skillset/react',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Jest',
          excerpt: 'It Jest needs to be done',
          favorite: true,
          icon: '',
          skillColor: 'blue',
          time: 1336,
        },
        generatedRoute: '/skillset/jest',
      },
    },
  ],
}

const mockProgrammingProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Portfolio',
          excerpt:
            'Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme',
          favorite: true,
          tags: [
            'Gatsby',
            'TypeScript',
            'React',
            'GraphQL',
            'GSAP',
            'Styled-Components',
            'Jest',
            'Enzyme',
          ],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Loopy',
          excerpt: 'I made a fantastic hula-loop once',
          favorite: true,
          tags: ['Hula', 'Loop', 'Perfection'],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
  ],
}

const mockGraphicDesignProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Circle',
          excerpt: 'I drew a fantastic circle once',
          favorite: true,
          tags: ['Pencil', 'Shapes', 'Drawing'],
        },
        generatedRoute: '/projects/circle',
      },
    },
  ],
}

const mockGraphicDesignSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Affinity Designer',
          excerpt: 'It has no proper geometry operations',
          favorite: false,
          icon: '',
          skillColor: 'green',
          time: 420,
        },
        generatedRoute: '/skillset/affinity_designer',
      },
    },
  ],
}

const mockExperienceMdx: ExperienceAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          title: 'React',
          backgroundColor: 'blue',
          type: 'Reactor',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'JSX is just fantastic',
        },
        generatedRoute: '',
      },
    },
    {
      node: {
        frontmatter: {
          title: 'Jest',
          backgroundColor: 'red',
          type: 'Jester',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'Jest a test',
        },
        generatedRoute: '',
      },
    },
  ],
}

export class StaticDataManager implements AStaticDataManager {
  static get theme(): DefaultTheme {
    return {
      palette: {
        dark: {
          normal: '#000',
          hover: '#222',
          active: '#444',
          lens: '#666',
        },
        light: {
          normal: '#fff',
          hover: '#ddd',
          active: '#bbb',
          lens: '#999',
        },
        first: { normal: '#f00', hover: '#d00', active: '#b00', lens: '#900' },
        second: {
          normal: '#0f0',
          hover: '#0d0',
          active: '#0b0',
          lens: '#090',
        },
        third: { normal: '#00f', hover: '#00d', active: '#00b', lens: '#009' },
        filter: {
          normal: 'saturation(4)',
          hover: 'saturation(4)',
          active: 'saturation(4)',
        },
      },
      breakpoints: {
        min: {
          smartphone: 'only screen and (min-width: 600px)',
          tablet: 'only screen and (min-width: 960px)',
          desktop: 'only screen and (min-width: 1200px)',
          ultrawide: 'only screen and (min-width: 1920px)',
        },
        max: {
          smartphone: 'only screen and (max-width: 600px)',
          tablet: 'only screen and (max-width: 960px)',
          desktop: 'only screen and (max-width: 1200px)',
          ultrawide: 'only screen and (max-width: 1920px)',
        },
      },
      fonts: {
        headings: 'Arial',
        copy: 'serif',
      },
    }
  }

  static get siteMetadata(): SiteMetadata {
    return {
      author: 'Jester',
      description: 'The TestRunner',
      image: '/',
      title: 'Jest',
      url: 'https://www.jest.testrunner',
    }
  }

  static get contacts(): Contact[] {
    return [
      {
        background: 'red',
        href: 'https://mock.test.nowhere',
        iconPath: '',
        text: 'Mock Option',
      },
    ]
  }

  static get courses(): Course[] {
    return [
      {
        category: 'Development',
        description: 'Experimenting with vanilla JS & jQuery',
        name: 'WebProgramming',
      },
    ]
  }

  static get routes(): RouteObject[] {
    return [
      {
        path: '/route',
        title: 'Route',
        boundKeys: ['R', 1],
      },
    ]
  }

  static get tools(): Tool[] {
    return [
      {
        name: 'React',
        svg: mockSVGs().RandomSampleSVG,
        color: 'red',
      },
    ]
  }

  static get photographyData(): PhotographyData {
    return {
      photographyData: mockImageCollection,
    }
  }

  static get photographySquares(): PhotographySquares {
    return {
      photographySquares: mockImageCollection,
    }
  }

  static get seriesPosters(): SeriesPosters {
    return {
      seriesPosters: mockImageCollection,
    }
  }

  static get gameWallpapers(): GameWallpapers {
    return {
      gameWallpapers: mockImageCollection,
    }
  }

  static get skillsetData(): SkillsetData {
    return {
      all: { ...mockProgrammingSkillsetMdx, ...mockGraphicDesignSkillsetMdx },
      top: {
        edges: [
          ...mockProgrammingSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
          ...mockGraphicDesignSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingSkillsetMdx,
      allGraphicDesign: mockGraphicDesignSkillsetMdx,
    }
  }

  static get projectsData(): ProjectsData {
    return {
      all: { ...mockGraphicDesignProjectsMdx, ...mockProgrammingProjectsMdx },
      top: {
        edges: [
          ...mockGraphicDesignProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
          ...mockProgrammingProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingProjectsMdx,
      allGraphicDesign: mockGraphicDesignProjectsMdx,
    }
  }

  static get experienceData(): ExperienceData {
    return {
      experienceData: mockExperienceMdx,
    }
  }

  static get dynamicIcons(): DynamicIcons {
    return {
      dynamicIcons: mockImageCollection,
    }
  }

  static get dynamicImages(): DynamicImages {
    return {
      dynamicImages: mockImageCollection,
    }
  }
}
