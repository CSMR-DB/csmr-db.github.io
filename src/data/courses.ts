import { theme } from './theme'

export interface ICourse {
  name: string
  description: string
  category:
    | 'Development'
    | 'Graphic Design'
    | 'Marketing'
    | 'Organizations'
    | 'Research'
    | 'Misc'
  color?: string
}

export const devCourses: ICourse[] = [
  {
    name: 'WebSkills',
    description: 'Just the absolute starter kit for working with HTML and CSS.',
    category: 'Development',
    color: theme.palette.dark.normal,
  },
  {
    name: 'Web Programming',
    description: `$('JavaScript'|'jQuery')`,
    category: 'Development',
    color: theme.palette.dark.normal,
  },
  {
    name: 'Realisation',
    description:
      'Using tools other than code to produce functional little apps. We worked with stuff like Appery.io and Layar.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
  {
    name: 'The Digital Workplace *',
    description:
      'A group project in about 8 groups of 4 to 5 students working together to create a prototype for a new Intranet for a client.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
  {
    name: 'Content Management Systems *',
    description: 'Building a website using Wordpress.',
    color: theme.palette.dark.normal,
    category: 'Development',
  },
]

export const marketingCourses: ICourse[] = [
  {
    name: 'Marketing & e-Business',
    description: 'An introduction course into digital marketing.',
    category: 'Marketing',
    // color: 'blue'
  },
  {
    name: 'Marketing & Communication',
    description: '',
    category: 'Marketing',
    // color: 'blue'
  },
  {
    name: 'Digital Methods',
    description:
      'Blogging about specific, current, topics regarding marketing.',
    category: 'Marketing',
    // color: 'blue'
  },
]

export const designCourses: ICourse[] = [
  {
    name: 'Workshop **',
    description:
      'We had 2 Workshops and these were not typical courses with an assessment, but instead were meant as an aid to other courses that relied on specific knowledge but were not designed to provide that in itself.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Interaction Design',
    description:
      'Learning about core design principles like the 5-7 rule, visual aligment and more.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Webdesign (with Axure)',
    description:
      'Apply what was taught at Interaction Design in a functional prototype built with Axure.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Digital Storytelling',
    description:
      'Use your knowledge about Adobe Photoshop, Illustrator, After Effects and whatever else you want to produce an animated short.',
    category: 'Graphic Design',
    // color: 'purple'
  },
  {
    name: 'Film',
    description: 'Working with a camera, as opposed to just digital assets.',
    category: 'Graphic Design',
    // color: 'purple'
  },
]

export const organizationCourses: ICourse[] = [
  {
    name: 'Organisations & Communication',
    description: '',
    category: 'Organizations',
    // color: 'blue'
  },
  {
    name: 'Organisations & Change Management',
    description:
      'Say you need to implement a different workflow or new tools in an organization. How would you go about doing so?',
    category: 'Organizations',
    // color: 'blue'
  },
  {
    name: 'Processes & Information Supply',
    category: 'Organizations',
    description: '',
    // color: 'red'
  },
]

export const researchCourses: ICourse[] = [
  {
    name: 'Evaluation Research',
    description: 'Focus on qualitative research.',
    category: 'Research',
    // color: 'red'
  },
  {
    name: 'Testing & Measuring',
    description: 'Focus on quantitative research',
    category: 'Research',
    // color: 'red'
  },
]

export const miscCourses: ICourse[] = [
  {
    name: 'Doing New Things',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Professional Skills',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Legal Skills',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
  {
    name: 'Entrepreneurship',
    description: '',
    category: 'Misc',
    // color: 'blue'
  },
]

export const courses: ICourse[] = [
  ...devCourses,
  ...designCourses,
  ...marketingCourses,
  ...organizationCourses,
  ...researchCourses,
  ...miscCourses,
]
