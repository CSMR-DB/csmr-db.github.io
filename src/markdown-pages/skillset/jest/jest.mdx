---
category: 'Programming'
level: 2
skillColor: '#C63D14'
time: 400
title: 'Jest'
excerpt: "I mean, it's one of many testing frameworks. I mainly just like the logo."
icon: <JestSVG/>
---
