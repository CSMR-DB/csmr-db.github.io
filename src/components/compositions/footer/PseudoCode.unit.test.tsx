import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'

import { RTRTree } from '../../../types/jest.types'
import { StaticDataManager } from '../../../data/StaticDataManager'

import { PseudoCode } from './PseudoCode'

describe('PseudoCode', (): void => {
  it('should match snapshot', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <PseudoCode />
      </ThemeProvider>
    )

    expect(comp).toMatchSnapshot()
  })

  it('should have the following static style rules', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <PseudoCode />
      </ThemeProvider>
    )
    const json: RTRTree = comp.toJSON()

    expect(comp).toMatchSnapshot()

    expect(json).toHaveStyleRule('margin-bottom', '1rem')

    /** Nested rules */
    expect(json).toHaveStyleRule('margin', '0', {
      modifier: '& > p',
    })
    expect(json).toHaveStyleRule('text-align', 'left', {
      modifier: '& > p',
    })
  })

  it('should render internally', (): void => {
    const comp: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>> = mount(
      <ThemeProvider theme={StaticDataManager.theme}>
        <PseudoCode />
      </ThemeProvider>
    )

    expect(comp.find('p').length).toBeGreaterThan(1)
  })
})
