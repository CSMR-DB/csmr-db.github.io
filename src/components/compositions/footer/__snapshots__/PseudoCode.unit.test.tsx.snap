// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PseudoCode should have the following static style rules 1`] = `
.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: white !important;
}

.c3:hover {
  color: #d00;
}

.c3.active {
  color: #b00;
}

.c0 {
  margin-bottom: 1rem;
}

.c0 > p {
  margin: 0;
  text-align: left;
}

.c1 {
  margin: 0 !important;
  padding-left: 0rem;
}

.c2 {
  margin: 0 !important;
  padding-left: 1rem;
}

.c4 {
  margin: 0 !important;
  padding-left: 2rem;
}

.c5 {
  margin: 0 !important;
  padding-left: 3rem;
}

.c6 {
  margin: 0 !important;
  padding-left: 4rem;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    type Requirements = 
    { essence: Emoji[] }
  </p>
  <p
    className="c1"
  >
    type DevWebsite = 
    { typescript: boolean }
  </p>
  <p
    className="c1"
  >
    interface IWebsite 
    {
  </p>
  <p
    className="c2"
  >
    url: string
  </p>
  <p
    className="c2"
  >
    visit: () =&gt; WebPage
  </p>
  <p
    className="c1"
  >
    }
  </p>
  <br />
  <p
    className="c1"
  >
    async function websitePromise({ essence }: Requirements): Promise&lt;IWebsite&gt; {
  </p>
  <p
    className="c2"
  >
    return await start(
    <a
      className="c3"
      href="https://www.gatsbyjs.org"
      target="blank"
    >
      Gatsby
    </a>
    )
  </p>
  <p
    className="c4"
  >
    .then((dev: DevWebsite) =
    &gt;
     
  </p>
  <p
    className="c5"
  >
    pipe(
  </p>
  <p
    className="c6"
  >
    addStyle(
    <a
      className="c3"
      href="https://www.styled-components.com/"
      target="blank"
    >
      styled-components
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    useFont(
    <a
      className="c3"
      href="https://www.npmjs.com/package/firacode"
      target="blank"
    >
      Fira Code
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    addCodeIcons(
    <a
      className="c3"
      href="https://marketplace.visualstudio.com/items?itemName=vscode-icons-team.vscode-icons"
      target="blank"
    >
      vscode-icons
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    addAdobeIcons(
    <a
      className="c3"
      href="https://www.iconfinder.com/graphiqa"
      target="blank"
    >
      Graphiqa Studio
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    prettifyCodeUsing(
    <a
      className="c3"
      href="https://prettier.io/"
      target="blank"
    >
      Prettier
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    makeCodeImagesUsing(
    <a
      className="c3"
      href="https://carbon.now.sh/"
      target="blank"
    >
      Carbon
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    useAnimationLibraries(
    <a
      className="c3"
      href="https://greensock.com/gsap/"
      target="blank"
    >
      GSAP
    </a>
    , 
    <a
      className="c3"
      href="https://www.npmjs.com/package/react-gsap"
      target="blank"
    >
      react-gsap
    </a>
    ,
     
    <a
      className="c3"
      href="https://www.gatsbyjs.org/packages/gatsby-plugin-transitions/"
      target="blank"
    >
      gatsby-plugin-transitions
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    sprinkleWith(essence),
  </p>
  <p
    className="c6"
  >
    deployUsing(
    <a
      className="c3"
      href="https://www.npmjs.com/package/gh-pages"
      target="blank"
    >
      gh-pages
    </a>
    )
  </p>
  <p
    className="c5"
  >
    )(dev)
  </p>
  <p
    className="c4"
  >
    )
  </p>
  <p
    className="c1"
  >
    }
  </p>
  <br />
  <p
    className="c1"
  >
    websitePromise(
    { essence: [ ‚ù§Ô∏è, ‚ú®, üéÜ ] }
    )
  </p>
  <p
    className="c2"
  >
    .then((website: IWebsite) =
    &gt;
  </p>
  <p
    className="c4"
  >
    website.visit())
  </p>
  <p
    className="c2"
  >
    )
  </p>
</div>
`;

exports[`PseudoCode should match snapshot 1`] = `
.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: white !important;
}

.c3:hover {
  color: #d00;
}

.c3.active {
  color: #b00;
}

.c0 {
  margin-bottom: 1rem;
}

.c0 > p {
  margin: 0;
  text-align: left;
}

.c1 {
  margin: 0 !important;
  padding-left: 0rem;
}

.c2 {
  margin: 0 !important;
  padding-left: 1rem;
}

.c4 {
  margin: 0 !important;
  padding-left: 2rem;
}

.c5 {
  margin: 0 !important;
  padding-left: 3rem;
}

.c6 {
  margin: 0 !important;
  padding-left: 4rem;
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    type Requirements = 
    { essence: Emoji[] }
  </p>
  <p
    className="c1"
  >
    type DevWebsite = 
    { typescript: boolean }
  </p>
  <p
    className="c1"
  >
    interface IWebsite 
    {
  </p>
  <p
    className="c2"
  >
    url: string
  </p>
  <p
    className="c2"
  >
    visit: () =&gt; WebPage
  </p>
  <p
    className="c1"
  >
    }
  </p>
  <br />
  <p
    className="c1"
  >
    async function websitePromise({ essence }: Requirements): Promise&lt;IWebsite&gt; {
  </p>
  <p
    className="c2"
  >
    return await start(
    <a
      className="c3"
      href="https://www.gatsbyjs.org"
      target="blank"
    >
      Gatsby
    </a>
    )
  </p>
  <p
    className="c4"
  >
    .then((dev: DevWebsite) =
    &gt;
     
  </p>
  <p
    className="c5"
  >
    pipe(
  </p>
  <p
    className="c6"
  >
    addStyle(
    <a
      className="c3"
      href="https://www.styled-components.com/"
      target="blank"
    >
      styled-components
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    useFont(
    <a
      className="c3"
      href="https://www.npmjs.com/package/firacode"
      target="blank"
    >
      Fira Code
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    addCodeIcons(
    <a
      className="c3"
      href="https://marketplace.visualstudio.com/items?itemName=vscode-icons-team.vscode-icons"
      target="blank"
    >
      vscode-icons
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    addAdobeIcons(
    <a
      className="c3"
      href="https://www.iconfinder.com/graphiqa"
      target="blank"
    >
      Graphiqa Studio
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    prettifyCodeUsing(
    <a
      className="c3"
      href="https://prettier.io/"
      target="blank"
    >
      Prettier
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    makeCodeImagesUsing(
    <a
      className="c3"
      href="https://carbon.now.sh/"
      target="blank"
    >
      Carbon
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    useAnimationLibraries(
    <a
      className="c3"
      href="https://greensock.com/gsap/"
      target="blank"
    >
      GSAP
    </a>
    , 
    <a
      className="c3"
      href="https://www.npmjs.com/package/react-gsap"
      target="blank"
    >
      react-gsap
    </a>
    ,
     
    <a
      className="c3"
      href="https://www.gatsbyjs.org/packages/gatsby-plugin-transitions/"
      target="blank"
    >
      gatsby-plugin-transitions
    </a>
    ),
  </p>
  <p
    className="c6"
  >
    sprinkleWith(essence),
  </p>
  <p
    className="c6"
  >
    deployUsing(
    <a
      className="c3"
      href="https://www.npmjs.com/package/gh-pages"
      target="blank"
    >
      gh-pages
    </a>
    )
  </p>
  <p
    className="c5"
  >
    )(dev)
  </p>
  <p
    className="c4"
  >
    )
  </p>
  <p
    className="c1"
  >
    }
  </p>
  <br />
  <p
    className="c1"
  >
    websitePromise(
    { essence: [ ‚ù§Ô∏è, ‚ú®, üéÜ ] }
    )
  </p>
  <p
    className="c2"
  >
    .then((website: IWebsite) =
    &gt;
  </p>
  <p
    className="c4"
  >
    website.visit())
  </p>
  <p
    className="c2"
  >
    )
  </p>
</div>
`;
